src/modules/product/product.service.js

```
@@ -384,6 +396,117 @@ export const getCategoryByIdService = async (id) => {
+  
+  // Sanitize pagination parameters
+  const page = Math.max(1, parseInt(options.page) || 1);
+  const limit = Math.max(1, Math.min(100, parseInt(options.limit) || 10));
+  
+  // Handle MongoDB sort syntax with hyphen prefix
+  let sortField = 'createdAt'; // Default field
+  let sortDirection = -1;      // Default direction (newest first)
+  
+  if (options.sortBy) {
+    // Check if sortBy starts with hyphen (indicating descending order)
+    if (options.sortBy.startsWith('-')) {
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)